autoprotocol/protocol.py

```diff
@@ -5135,10 +5141,30 @@ def provision(self, resource_id, dests, volumes):
                 self.instructions[-1].to.append(xfer)
             else:
                 provision_instructions_to_return.append(
-                    self._append_and_return(Provision(resource_id, dest_group))
+                    self._append_and_return(
+                        Provision(resource_id, dest_group, measurement_mode)
+                    )
                 )
         return provision_instructions_to_return
 
+    def _identify_measurement_mode(self, volumes):
+        unique_measure_modes = set()
+        for amount in volumes:
+            if not isinstance(amount, (str, Unit)):
+                raise TypeError(f"Provided amount {amount} is not supported.")
+            if amount.unit.endswith("liter"):
```

Comment:
@yangchoo & @joshhacksthings 
I prefer using below as it will be more generic:
```
# For mass
amount.dimensionality == Unit(1, "gram").dimensionality 
# For volume
amount.dimensionality == Unit(1, "liter").dimensionality
```
Modified the code accordingly